cli/clibase/values.go

```diff
@@ -24,6 +24,49 @@ type NoOptDefValuer interface {
 	NoOptDefValue() string
 }
 
+// Validator is a wrapper around a pflag.Value that allows for validation
+// of the value after or before it has been set.
+type Validator[T pflag.Value] struct {
+	Value T
+	// validate is called after the value is set.
+	validate func(T) error
+}
+
+func Validate[T pflag.Value](opt T, validate ...func(value T) error) *Validator[T] {
```

Comment:
What's the use-case for variadic validate funcs? Do we foresee passing multiple funcs? Typically I'm all for it, but with generics it seems to hurt VSCode autocomplete-ability (autocompletes to `[]func` 😒).

Reply:
Eww on the autocomplete.

The only rational was a comment by @johnstcn that it might make sense to create building block style validators that you could reuse. But I can put it back to a regular, and then we can create a `CombineValidator` or something that allows combining. 

I had no strong opinions either way.

Reply:
Huh my autocomplete works fine in Goland. Still we can make the varadic some helper func. Or make it a variadic when we need that